gtp_engine.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Version of the GTP specification used:
+#   https://www.lysator.liu.se/~gunnar/gtp/gtp2-spec-draft2/gtp2-spec.html
+
+import inspect
+import re
+import sys
+import traceback
+
+
+# List of canonical error messages as required by 6.3 in the specification.
+_CANONICAL_ERRORS = {
```

Comment:
Should these be referenced in e.g the ValueError raised in `cmd_boardsize`?  might be DRY-er

Reply:
I've updated the exceptions to contain more meaningful information